tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/types/json/BJSONValueTest.java

```diff
@@ -462,4 +462,40 @@ public void testJsonIntToFloat() {
         Assert.assertNotNull(returns[0]);
         Assert.assertEquals(((BFloat) returns[0]).floatValue(), 4.0);
     }
+
+    @Test(expectedExceptions = { BLangRuntimeException.class },
+            expectedExceptionsMessageRegExp = "error: error, message: 'null' cannot be cast to 'int'.*")
+    public void testNullJsonToInt() {
+        BRunUtil.invoke(compileResult, "testNullJsonToInt");
+    }
+
+    @Test(expectedExceptions = { BLangRuntimeException.class },
+            expectedExceptionsMessageRegExp = "error: error, message: 'null' cannot be cast to 'float'.*")
+    public void testNullJsonToFloat() {
+        BRunUtil.invoke(compileResult, "testNullJsonToFloat");
+    }
+
+    @Test(expectedExceptions = { BLangRuntimeException.class },
+            expectedExceptionsMessageRegExp = "error: error, message: 'null' cannot be cast to 'string'.*")
+    public void testNullJsonToString() {
+        BRunUtil.invoke(compileResult, "testNullJsonToString");
+    }
+
+    @Test(expectedExceptions = { BLangRuntimeException.class },
+            expectedExceptionsMessageRegExp = "error: error, message: 'null' cannot be cast to 'boolean'.*")
+    public void testNullJsonToBoolean() {
+        BRunUtil.invoke(compileResult, "testNullJsonToBoolean");
+    }
+
+    @Test(expectedExceptions = { BLangRuntimeException.class },
+            expectedExceptionsMessageRegExp = "error: ballerina/runtime:NullReferenceException.*")
```

Comment:
This seems to be wrong. the error should be "'null' cannot be cast to "Type".  Idealy we should remove NullReferenceException as well.

Reply:
+1